### ISSUE TITLE
Incorrectly reports "The caught exception is swallowed." finding

### ISSUE THREAD

**Issue by bay73** [detekt/detekt#7928]
## Expected Behavior

Code block
```
        try {
            createUser(attributes)
        } catch (e: InvalidParameterException) {
            logger.warn("Failed to create user due to invalid parameters", e)
            throw ConstraintViolationException( e.message ?: "Failed to create user due to invalid attributes")
        }
```
Shouldn't report swallowed exception.

## Observed Behavior

Executing detektion raises `The caught exception is swallowed. The original exception could be lost. [SwallowedException]`
The interesting thing that if I remove `e.message` from the throw expression, then it doesn't report anything.

## Your Environment
* Version of detekt used: 1.23.7
* Version of Gradle used (if applicable): 8.12.1
* Operating System and version: macOS Sequoia 15.1

**Comment by BraisGabin** [comment]
It is swallowed. Like this it will be not swallowed:

```kotlin
throw ConstraintViolationException(e.message ?: "Failed to create user due to invalid attributes", e)
```

You can read the documentation here and how your code is in the "Noncompliant code" section. https://detekt.dev/docs/rules/exceptions/#swallowedexception

I'm closing but if you still thinks that the rule isn't doing what it is expected please comment back :)
